DeepSpeech.py

```diff
@@ -694,594 +651,999 @@ def stopwatch(start_duration=0):
+                    # Check, if epoch is done now
+                    if epoch.done():
+                        # If it declares itself done, move it from 'running' to 'done' collection
+                        self._epochs_running.remove(epoch)
+                        self._epochs_done.append(epoch)
+                        # Show the short and/or full WER report
+                        log_info(epoch)
+            else:
+                # There was no running epoch found for this job - this should never happen.
+                log_error('There is no running epoch of id %d for job with ID %d.' % (job.epoch_id, job.id))
+            return self.get_job(job.worker)
+        else:
+            # We are a remote worker and have to hand over to the chief worker by HTTP
+            result = self._talk_to_chief('', data=pickle.dumps(job))
+            if result:
+                result = pickle.loads(result)
+            return result
+
+
+def read_data_sets(set_names=['train', 'dev', 'test']):
+    r'''
+    Returns a :class:`DataSets` object of the selected importer, containing all available/selected sets.
+    '''
+    return importer_module.read_data_sets(FLAGS.dataset_path,
+                                          FLAGS.train_batch_size,
+                                          FLAGS.dev_batch_size,
+                                          FLAGS.test_batch_size,
+                                          n_input,
+                                          n_context,
+                                          next_index=lambda set_name, index: COORD.get_next_index(set_name),
+                                          limit_dev=FLAGS.limit_dev,
+                                          limit_test=FLAGS.limit_test,
+                                          limit_train=FLAGS.limit_train,
+                                          sets=set_names)
+
+def train(server=None):
+    r'''
+    Trains the network on a given server of a cluster.
+    If no server provided, it performs single process training.
+    '''
+
+    # Create a variable to hold the global_step.
+    # It will automgically get incremented by the optimizer.
+    global_step = tf.Variable(0, trainable=False, name='global_step')
+
+    # Read all data sets
+    data_sets = read_data_sets()
+
+    # Get the data sets
+    switchable_data_set = SwitchableDataSet(data_sets)
+
+    # Create the optimizer
+    optimizer = create_optimizer()
+
+    # Synchronous distributed training is facilitated by a special proxy-optimizer
+    if not server is None:
+        optimizer = tf.train.SyncReplicasOptimizer(optimizer,
+                                                   replicas_to_aggregate=FLAGS.replicas_to_agg,
+                                                   total_num_replicas=FLAGS.replicas)
+
+    # Get the data_set specific graph end-points
+    tower_results = get_tower_results(switchable_data_set, optimizer)
+    results_tuple, gradients, accuracy, loss = tower_results
```

Comment:
You can do this inline, since we don't use `tower_results` directly anywhere:
```python
    results_tuple, gradients, accuracy, loss = get_tower_results(switchable_data_set, optimizer)
```